modules/matchRoutes.js

```diff
@@ -35,9 +35,7 @@ function getIndexRoute(route, location, callback) {
       callback(error, !error && createRoutes(indexRoute)[0])
     })
   } else if (route.childRoutes) {
-    const pathless = route.childRoutes.filter(function (obj) {
-      return !obj.hasOwnProperty('path')
-    })
+    const pathless = route.childRoutes.filter(childRoute => !childRoute.path)
```

Comment:
This is just a cleaner way to do that check. I considered writing this as

```js
({ path }) => !path
```

but thought using the explicit `childRoute` as the argument name was clearer.